cibyl/_data/schemas/config.json

```diff
@@ -0,0 +1,193 @@
+                  "type": "string",
+                  "enum": [
+                    "jenkins",
+                    "zuul"
+                  ]
+                },
+                "sources": {
+                  "type": "object",
+                  "additionalProperties": false,
+                  "patternProperties": {
+                    "^.*$": {
+                      "type": "object",
+                      "oneOf": [
+                        {
+                          "$ref": "#/$defs/jenkins"
+                        },
+                        {
+                          "$ref": "#/$defs/jjb"
+                        },
+                        {
+                          "$ref": "#/$defs/es"
+                        },
+                        {
+                          "$ref": "#/$defs/zuul"
+                        },
+                        {
+                          "$ref": "#/$defs/zuul.d"
+                        }
+                      ]
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "plugins": {
+      "type": "array",
+      "additionalItems": false,
+      "items": {
+        "type": "string"
+      }
+    }
+  },
+  "required": [
+    "environments"
+  ],
+  "$defs": {
+    "jenkins": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "driver": {
+          "type": "string",
+          "const": "jenkins"
+        },
+        "url": {
+          "type": "string"
+        },
+        "username": {
+          "type": "string"
+        },
+        "token": {
+          "type": "string"
+        },
+        "cert": {
+          "type": "boolean"
+        },
+        "enabled": {
+          "type": "boolean"
+        }
+      }
+    },
+    "jjb": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "driver": {
+          "type": "string",
+          "const": "jenkins_job_builder"
+        },
+        "repos": {
+          "type": "array",
+          "additionalItems": false,
+          "items": {
+            "type": "object",
+            "additionalProperties": false,
+            "properties": {
+              "url": {
+                "type": "string"
+              }
+            }
+          }
+        },
+        "enabled": {
+          "type": "boolean"
+        }
+      }
+    },
+    "es": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "driver": {
+          "type": "string",
+          "const": "elasticsearch"
+        },
+        "url": {
+          "type": "string"
+        },
+        "enabled": {
+          "type": "boolean"
+        }
+      }
+    },
+    "zuul": {
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "driver": {
+          "type": "string",
+          "const": "zuul"
+        },
+        "url": {
```

Comment:
You'll know better than me, but `driver` and `url` should be required for zuul

Reply:
Done.